src/ralph/backends/data/s3.py

```diff
@@ -0,0 +1,386 @@
+            chunk_size (int or None): Ignored.
+            ignore_errors (bool): If `True`, errors during the write operation
+                are ignored and logged. If `False` (default), a `BackendException`
+                is raised if an error occurs.
+            operation_type (BaseOperationType or None): The mode of the write
+                operation.
+                If operation_type is `CREATE` or `INDEX`, the target object is
+                expected to be absent. If the target object exists a
+                `BackendException` is raised.
+
+        Return:
+            int: The number of written objects.
+
+        Raise:
+            BackendException: If a failure during the write operation occurs.
+            BackendParameterException: If a backend argument value is not valid.
+        """
+        data = iter(data)
+        try:
+            first_record = next(data)
+        except StopIteration:
+            logger.info("Data Iterator is empty; skipping write to target.")
+            return 0
+
+        if not operation_type:
+            operation_type = self.default_operation_type
+
+        if not target:
+            target = f"{self.default_bucket_name}/{now()}-{uuid4()}"
+            logger.info(
+                "Target not specified; using default bucket with random file name: %s",
+                target,
+            )
+
+        elif "/" not in target:
+            target = f"{self.default_bucket_name}/{target}"
+            logger.info(
+                "Target not specified; using default bucket: %s",
+                target,
+            )
+
+        target_bucket, target_object = target.split("/", 1)
+
+        if operation_type in [
+            BaseOperationType.APPEND,
+            BaseOperationType.DELETE,
+            BaseOperationType.UPDATE,
+        ]:
+            msg = "%s operation_type is not allowed."
+            logger.error(msg, operation_type.name)
+            raise BackendParameterException(msg % operation_type.name)
+
+        if target_object in list(self.list(target=target_bucket)):
+            msg = "%s already exists and overwrite is not allowed for operation %s"
+            logger.error(msg, target_object, operation_type)
+            raise BackendException(msg % (target_object, operation_type))
+
+        logger.info("Creating archive: %s", target_object)
+
+        data = chain((first_record,), data)
+        if isinstance(first_record, dict):
+            data = self._parse_dict_to_bytes(data, ignore_errors)
+
+        counter = {"count": 0}
+        data = self._count(data, counter)
+
+        # Using StreamingIterator from requests-toolbelt but without specifying a size
+        # as we will not use it. It implements the `read` method for iterators.
+        data = StreamingIterator(0, data)
+
+        try:
+            self.client.upload_fileobj(
+                Bucket=target_bucket,
+                Key=target_object,
+                Fileobj=data,
+            )
```

Comment:
It seems the [upload_fileobj](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/upload_fileobj.html) method can be configured with a [TransferConfig](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/customizations/s3.html#boto3.s3.transfer.TransferConfig) object including a `multipart_chunksize` parameter similar to our `chunk_size` parameter.